### PR TITLE
Reorganise spec to construct varaibles in the scenario

### PR DESCRIPTION
**Pull request by weiqingtoh**
Arising from https://github.com/Coursemology/coursemology2/pull/1110#discussion_r68158327

### CODE REVIEW

File: spec/features/course/assessment/submissions_viewing_spec.rb [hunk 1]
```diff
@@ -10,22 +10,16 @@
...
       scenario 'I can view all submitted and graded submissions' do
+        students = create_list(:course_student, 3, :approved, course: course)
+        attempting_submission = create(:course_assessment_submission, :attempting,
+                                       assessment: assessment, creator: students[0].user)
+        submitted_submission  = create(:course_assessment_submission, :submitted,
+                                       assessment: assessment, creator: students[1].user)
+        graded_submission = create(:course_assessment_submission, :graded,
+                                   assessment: assessment, creator: students[2].user)
+
         visit course_submissions_path(course)
```
Comment (kxmbrian) [inline spec/features/course/assessment/submissions_viewing_spec.rb]: It is DRYer to do this? :
```ruby
attempting_submission, submitted_submission, graded_submission = 
  students.zip([:attempting, :submitted, :graded]).map do |student, state|
    create(:course_assessment_submission, state, assessment: assessment, creator: students.user)
  end
```

Comment (weiqingtoh) [inline spec/features/course/assessment/submissions_viewing_spec.rb]: Good point will try to DRY up the specs a little 👍

**Comment by kxmbrian** [comment]
👍

File: spec/features/course/assessment/submissions_viewing_spec.rb [hunk 2]
```diff
@@ -10,22 +10,15 @@
...
-      let(:graded_submission) do
-        create(:course_assessment_submission, :graded, assessment: assessment,
-                                                       creator: students[2].user)
-      end
-      let!(:submissions) { [attempting_submission, submitted_submission, graded_submission] }
 
       scenario 'I can view all submitted and graded submissions' do
+        students = create_list(:course_student, 3, :approved, course: course)
+        attempting_submission, submitted_submission, graded_submission =
+          students.zip([:attempting, :submitted, :graded]).map do |student, trait|
```
Comment (allenwq) [inline spec/features/course/assessment/submissions_viewing_spec.rb]: @weiqingtoh How does rubocop say ?

**Pull request merged**